framer/Layer.coffee

```diff
@@ -366,12 +362,16 @@ class exports.Layer extends BaseClass
 		get: ->
 			@_getPropertyValue("shadows")
 		set: (value) ->
-			shadows = _.cloneDeep(value)
-			for shadow in shadows
-				_.defaults shadow, Framer.Defaults.Shadow
-				shadow.color = new Color(shadow.color)
+			shadows = []
+			for shadow, index in value
+				shadows.push proxiedShadowValue(@, shadow, index)
 			@_setPropertyValue("shadows", shadows)
-			updateShadowStyle(@)
+			@updateShadowStyle()
+
+	updateShadowStyle: ->
```

Comment:
Well, you might want to call it if you manually change the shadows in the `layer.shadows` array